frontend/pages/admin/UserManagementPage/components/SelectedTeamsForm/SelectedTeamsForm.tsx

```diff
@@ -0,0 +1,95 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+
+import ITeam from 'interfaces/team';
+// ignore TS error for now until these are rewritten in ts.
+// @ts-ignore
+import Checkbox from 'components/forms/fields/Checkbox';
+// @ts-ignore
+import Dropdown from 'components/forms/fields/Dropdown';
+
+interface ICheckboxListItem {
+  name: string;
+  id: number;
+  isChecked: boolean | undefined;
+}
+
+interface ISelectedTeamsFormProps {
+  availableTeams: ITeam[];
+  selectedTeams: ITeam[];
+  onFormChange: (teams: ITeam[]) => void;
+}
+
+const baseClass = 'selected-teams-form';
+
+const roles = [
+  {
+    label: 'observer',
+    value: 'observer',
+  },
+  {
+    label: 'maintainer',
+    value: 'maintainer',
+  },
+];
+
+const generateListItems = (allTeams: ITeam[], selectedTeams: ITeam[]): ICheckboxListItem[] => {
+  if (selectedTeams.length === 0) {
+    return allTeams.map((team) => {
+      const { name, id } = team;
+      return {
+        name,
+        id,
+        isChecked: false,
+      };
+    });
+  }
+
+  // TODO: add functionality editing for selected teams.
+  return [];
+};
+
+const SelectedTeamsForm = (props: ISelectedTeamsFormProps): JSX.Element => {
+  const { availableTeams, selectedTeams, onFormChange } = props;
+  const checkboxListItems = generateListItems(availableTeams, selectedTeams);
+
+  const onChangeInput = (checkboxItem: ICheckboxListItem): void => {
+    const updatedTeam = selectedTeams.find(team => team.id === checkboxItem.id);
+
+    // TODO: figure out logic of newly checked and updated checked.
+    // if (updatedTeam === undefined) {
+    //
+    // } else {
+    //
+    // }
+    // onFormChange(selectedTeam);
+  };
```

Comment:
I'm not sure we will need to keep track of which were updated, but we can just track the current state and upload the whole thing in the request. Does that make sense? Not sure if I'm interpreting your comment correctly here.

Reply:
This changed. The approach I was taking before didnt make sense. Now the SelectedTeamsForm take in a list off all teams, a list of all teams a user is in, and spits out form data of the selected teams on the roles for them.